### PR TITLE
Add support for partial methods in interfaces

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs [hunk 1]
```diff
@@ -8583,6 +8583,554 @@ static void Main()
...
+            var compilation2 = CreateCompilation(source1 + source2, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.Regular,
+                                                 targetFramework: TargetFramework.DesktopLatestExtended);
+
+            compilation2.VerifyDiagnostics(
+                // (9,18): error CS8701: Target runtime doesn't support default interface implementation.
+                //     partial void M1() {}
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, "M1").WithLocation(9, 18)
+                );
+        }
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: Consider tests when an explicit access modifier is used on a `partial` method.

Comment (jaredpar) [reply src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: Consider a test where a non-void return is used.

---
In reply to: [259462424](https://github.com/dotnet/roslyn/pull/33594#discussion_r259462424) [](ancestors = 259462424)

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs]: Both covered in MethodModifiers_26

---
In reply to: [259462522](https://github.com/dotnet/roslyn/pull/33594#discussion_r259462522) [](ancestors = 259462522,259462424)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off

**Pull request merged**